azure-keyvault/porter.yaml

```diff
@@ -0,0 +1,157 @@
+name: hello-keyvault
+version: 0.1.2
+description: "Creates and azure keyvault using the az cli"
+registry: ghcr.io/asalbers
+dockerfile: Dockerfile.tmpl
+
+mixins:
+  - exec
+  - az
+
+install:
+  - az: 
+      description: "Logging into Azure..."  
+      arguments:
+        - login
+      suppress-output: true
+      flags:
+        debug: ""
+        service-principal:
+        username: "'{{bundle.credentials.AZURE_CLIENT_ID}}'"
+        password: "'{{bundle.credentials.AZURE_SP_PASSWORD}}'"
+        tenant: "'{{bundle.credentials.AZURE_TENANT_ID_OR_DNS}}'"
+  - az: 
+      description: "Setting the Azure subscription...."
+      suppress-output: true
+      arguments: 
+        - "account" 
+        - "set" 
+      flags:
+        subscription: "{{ bundle.credentials.AZURE_SUBSCRIPTION_ID}}"
+  - exec:
+      command: bash
+      description: "Creating or using the Azure resource group...."
+      arguments:
+        - ./upsertGroup.sh
+        - "{{bundle.parameters.AZURE_RESOURCE_GROUP_NAME}}"
+        - "{{bundle.parameters.AZURE_GROUP_REGION}}"
+  - exec:
+      command: bash
+      description: "Setting random string...."
+      arguments:
+        - ./helpers.sh
+        - random_string
+      outputs:
+        - name: random_string
+          regex: "(.*)"
+  - az:
+      description: "Creating the KeyVault..."
+      suppress-output: true
+      arguments:
+        - keyvault 
+        - create 
+      flags:
+        location: "eastus2"
+        name: "test-vault-{{bundle.outputs.random_string}}"
+        resource-group: "{{bundle.parameters.AZURE_RESOURCE_GROUP_NAME}}"
+        enable-rbac-authorization: "true"
+      outputs:
+        - name: "keyvault_name"
+          jsonPath: "$.name"
+
+upgrade:
```

Comment:
Just a note that the upgrade action isn't required by Porter, so you can always omit it if you don't have any upgrade logic.